### ISSUE TITLE
Error installing Capsule on Kind cluster: connection refused when trying to retrieve Capsule configuration

### ISSUE THREAD

**Issue by smileisak** [clastix/capsule#677]
# Bug description

There is an error when installing the Capsule on a fresh Kind cluster (version 1.24.1) using the command 
```bash
kubectl apply -f https://raw.githubusercontent.com/clastix/capsule/master/config/install.yaml.
``` 
The controller log shows a panic message stating that the Capsule configuration (named "capsule-default") cannot be retrieved due to a conversion webhook failure, caused by a connection refused error when trying to connect to the address "[https://capsule-webhook-service.capsule-system.svc:443/convert?timeout=30s](https://capsule-webhook-service.capsule-system.svc/convert?timeout=30s)". The default CapsuleConfiguration resource was created, but it appears to be missing certain overrides and may not be properly configured.

```bash
$ k get capsuleconfigurations.capsule.clastix.io  capsule-default -o yaml
apiVersion: capsule.clastix.io/v1beta2
kind: CapsuleConfiguration
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"capsule.clastix.io/v1beta2","kind":"CapsuleConfiguration","metadata":{"annotations":{},"name":"capsule-default"},"spec":{"enableTLSReconciler":true,"forceTenantPrefix":false,"protectedNamespaceRegex":"","userGroups":["capsule.clastix.io"]}}
  creationTimestamp: "2023-01-03T15:58:45Z"
  generation: 1
  name: capsule-default
  resourceVersion: "1164"
  uid: f3ead9af-4cb8-4e04-b494-f1a0fb76c145
spec:
  enableTLSReconciler: true
  forceTenantPrefix: false
  overrides:
    TLSSecretName: capsule-tls
    mutatingWebhookConfigurationName: capsule-mutating-webhook-configuration
    validatingWebhookConfigurationName: capsule-validating-webhook-configuration
  protectedNamespaceRegex: ""
  userGroups:
  - capsule.clastix.io
```

# How to reproduce

1. Create a kind cluster: 
```bash
$ kind create cluster --image kindest/node:v1.24.1 --wait 5m --name capsule
```
2. Apply the install manifest from the documentation:
```bash
$ kubectl apply -f https://raw.githubusercontent.com/clastix/capsule/master/config/install.yaml
```

# Expected behavior

The expected behavior for this bug is for the Capsule to be successfully installed on the Kind cluster without any errors or issues. This includes the creation and proper configuration of the default CapsuleConfiguration resource, as well as the successful execution of any necessary webhooks. The installation process should complete without any panics or connection issues.

# Logs

```
2023-01-03T16:01:30.291Z	INFO	setup	Capsule Version v0.1.3 eba072c
2023-01-03T16:01:30.291Z	INFO	setup	Build from: https://github.com/clastix/capsule
2023-01-03T16:01:30.291Z	INFO	setup	Build date: 2022-12-02T14:34:21
2023-01-03T16:01:30.291Z	INFO	setup	Go Version: go1.18.8
2023-01-03T16:01:30.291Z	INFO	setup	Go OS/Arch: linux/amd64
2023-01-03T16:01:30.546Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
panic: Cannot retrieve Capsule configuration with name capsule-default: conversion webhook for capsule.clastix.io/v1beta2, Kind=CapsuleConfiguration failed: Post "https://capsule-webhook-service.capsule-system.svc:443/convert?timeout=30s": dial tcp 10.96.224.115:443: connect: connection refused

goroutine 1 [running]:
github.com/clastix/capsule/pkg/configuration.NewCapsuleConfiguration.func1()
	/workspace/pkg/configuration/client.go:41 +0x1f9
github.com/clastix/capsule/pkg/configuration.capsuleConfiguration.EnableTLSConfiguration({0x1815e48?})
	/workspace/pkg/configuration/client.go:82 +0x22
main.main()
	/workspace/main.go:145 +0xa11
```
# Additional context

- Capsule version:  `v0.1.3`
- Kubernetes version: `v1.24.1`

**Comment by smileisak** [comment]
It appears that the issue is related to the controller not being able to retrieve the TLS configuration when installing the Capsule using the manifest file. It seems that the secrets necessary for the TLS configuration are being created, but they are empty, which is causing the controller to fail when it attempts to retrieve the TLS certificates.

Alternatively, if installing the Capsule using the Helm chart has been successful, it may be helpful to use the Helm chart as the primary method for installing the Capsule, as it includes a post-job that generates the TLS configuration data and helps to ensure that the controller can retrieve it successfully.

**Comment by oliverbaehler** [comment]
Hi @smileisak 

Thanks for the interest in the project.
We are right now in the process of introducing a new API Version. We have verified all of our changes (also with our e2e testing) against the Helm installation.

We recommend Installation by Helm Chart. Can you try if that works for you?

I will take a look after we have done the new release, what the problem is with the install.yaml :)

**Comment by prometherion** [comment]
The problem here is that the single YAML manifest (`install.yaml`) is taken directly from the main branch which is addressing several changes and enhancements.

The image referenced in that manifest (v0.1.3) is not aware of the changes requested and addressed in the CRD that is evolving on that branch, thus, Capsule tries to request a conversion from CapsuleConfiguration `v1beta2` to `v1alpha1`, and the CapsuleConfiguration is required at startup time to start the Operator with the required settings.

I would suggest @smileisak install from the single YAML manifest using a permalink referenced to a version such as the [v0.1.3](https://raw.githubusercontent.com/clastix/capsule/v0.1.3/config/install.yaml) or by using the Helm chart which is versioned properly and way much more tested.

Anyway, if you want to give a try to the latest changes, you would need to build locally Capsule from the `master` branch and upload the resulting artefact to the `kind` cluster by issuing `kind load docker-image ...`.

I'll check if in the documentation we're referring to the single YAML manifest and, in case of direct `master` permalink, I'll let point it to a stable version.

**Comment by smileisak** [comment]
Updating the documentation to include this information could be helpful for others who may encounter this issue in the future. It may also be helpful to consider adding this annotation to the manifest file by default, to ensure that the required certificates are always generated and the webhook functions properly when installing Capsule using the manifest file.

**Comment by prometherion** [comment]
With the latest changes that are going to be released with v0.2.0, and the related promotion of CapsuleConfiguration to `v1beta2`, the issue is no anymore showing.
